src/main/groovy/com/google/protobuf/gradle/GenerateProtoTask.groovy

```diff
@@ -361,7 +361,8 @@ public class GenerateProtoTask extends DefaultTask {
     Preconditions.checkState(state == State.FINALIZED, 'doneConfig() has not been called')
 
     ToolsLocator tools = project.protobuf.tools
-    Set<File> protoFiles = inputs.sourceFiles.files
+    // Sort to make sure files are in a consistent order
```

Comment:
This comment says what you are doing and the obvious implication, but not why. Maybe:
> Sort to ensure generated descriptors have a canonical representation to avoid triggering unnecessary rebuilds downstream